src/hooks/useAsyncStorage.tsx

```diff
@@ -0,0 +1,27 @@
+import { useCallback, useEffect, useState } from 'react';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+export function useAsyncStorage(key: string, enabled: boolean = true) {
+  const [value, setValue] = useState<string | null>(null);
+
+  useEffect(() => {
+    (async () => {
+      if (enabled) {
+        const storedValue = await AsyncStorage.getItem(key);
+        setValue(storedValue);
```

Comment:
We used to have quite a bit of flow control in the app that relied on react-query's isLoading/isFetched conditions here. Instead we can just rely on the state update to trigger a re-render.